src/plugins/intel_npu/tests/functional/behavior/npuw/behavior_tests.cpp

```diff
@@ -53,209 +61,470 @@ TEST_F(BehaviorTestsNPUW, TestInfrastructureIsCorrect) {
+        EXPECT_COMPILE_MODEL(npu, 1);
+        EXPECT_COMPILE_MODEL(cpu, 1);
     }
-    npu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
-        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
-    });
-    npu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
-        EXPECT_CALL(request, infer())
-            .Times(1)
-            .WillOnce(Throw(std::runtime_error("Infer on MockNPU is failed")));
-    });   
-    cpu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
-        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
-    });
-    cpu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
-        EXPECT_CALL(request, infer()).Times(1);
-    });
+    EXPECT_CREATE_SYNC_INFER_REQ(npu, MODEL(0), 1);
+    EXPECT_INFER(npu, MODEL(0), 1, THROW("Infer on MockNPU is failed"));
+    EXPECT_CREATE_SYNC_INFER_REQ(cpu, MODEL(0), 1);
+    EXPECT_INFER(cpu, MODEL(0), 1);
 
     // Register mock objects as plugins in OpenVINO:
     register_mock_plugins_in_ov();
 
     // Do the actual test:
-    auto npuw_properties = 
-        ov::AnyMap{ov::intel_npu::use_npuw(true),
-                   ov::intel_npu::npuw::devices("MockNPU,MockCPU")};
-    auto compiled_model = core.compile_model(model, "NPU", npuw_properties);
+    use_npuw_props.emplace(devices("MockNPU,MockCPU"));
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
     auto infer_request = compiled_model.create_infer_request();
     EXPECT_NO_THROW(infer_request.infer());
 }
 
 TEST_F(BehaviorTestsNPUW, InferIsFailed) {
+    model = model_generator.get_model_with_one_op();
+
+    // Set expectations first:
+    EXPECT_COMPILE_MODEL(npu, 1);
+    EXPECT_COMPILE_MODEL(cpu, 0);
+
+    EXPECT_CREATE_SYNC_INFER_REQ(npu, MODEL(0), 1);
+    EXPECT_INFER(npu, MODEL(0), 1, THROW("Infer on MockNPU is failed"));
+
+    // Register mock objects as plugins in OpenVINO:
+    register_mock_plugins_in_ov();
+
+    // Do the actual test:
+    use_npuw_props.emplace(devices("MockNPU"));
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
+    auto infer_request = compiled_model.create_infer_request();
+    EXPECT_ANY_THROW(infer_request.infer());
+}
+
+using BehaviorTestsNPUWOnlinePartitioning = BehaviorTestsNPUW;
+TEST_F(BehaviorTestsNPUWOnlinePartitioning, CompilationIsSuccessful) {
+    const auto& model = model_generator.get_model_with_repeated_blocks();
+
+    // Set expectations first:
+    EXPECT_COMPILE_MODEL(npu, Between(2, 12));
+    EXPECT_COMPILE_MODEL(cpu, 0);
+
+    // Register mock objects as plugins in OpenVINO:
+    register_mock_plugins_in_ov();
+
+    // Do the actual test:
+    use_npuw_props.emplace(devices("MockNPU"));    
+    use_npuw_props.emplace(partitioning::online::min_size(12));
+    use_npuw_props.emplace(partitioning::online::keep_blocks(9));
```

Comment:
I believe these can go into the default configuration. I'd work fine for either of our models

Reply:
`keep_blocks` is removed, default configuration for all tests contains only `NPU_USE_NPUW`, then tests are extended to contain test-specific options